src/cmd/troubleshoot/checks.go

```diff
@@ -0,0 +1,85 @@
+package troubleshoot
+
+import (
+	"strings"
+
+	tserrors "github.com/Dynatrace/dynatrace-operator/src/cmd/troubleshoot/errors"
+	"github.com/Dynatrace/dynatrace-operator/src/functional"
+)
+
+type Result int
+
+const (
+	PASSED Result = iota + 1
+	FAILED
+	SKIPPED
+)
+
+type troubleshootFunc func(troubleshootCtx *troubleshootContext) error
+
+type Check struct {
+	Do            troubleshootFunc
+	Prerequisites []*Check
+	Name          string
+}
+
+type ChecksResults struct {
+	check2Result map[*Check]Result
+}
```

Comment:
If it has only one field, could this be 
```suggestion
type ChecksResults map[*Check]Result
```
instead?